### DISCUSSION TITLE
Mac M1

### DISCUSSION THREAD

**Discussion by MELKOHEN** [LizardByte/community#208] [Sunshine]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

Hello everyone, I am desperately trying to install Sunshine on Mac M1 but I am encountering several difficulties. The most common one is that the libraries are not compatible with arm. When I try to use macport, the command returns "command not found sunshine". Does the software work on Mac M1 with macOS Ventura? Thank you.

### Host Operating System

macOS

### Operating System Version

MacOs Ventura

### Architecture

arm

### Sunshine commit or version

Last

### Package

macOS - dmg

### GPU Type

none (software encoding)

### GPU Model

M1 Mac

### GPU Driver/Mesa Version

M1 Mac

### Config

```Shell
dyld[42223]: Library not loaded: /opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib
  Referenced from: <8648640B-CE33-3DF9-8EC7-34E841FCE43F> /opt/local/bin/sunshine-0.19.1
  Reason: tried: '/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (no such file), '/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libboost_locale-mt.dylib' (no such file), '/usr/lib/libboost_locale-mt.dylib' (no such file, not in dyld cache)
zsh: abort      sunshine
```

### Relevant log output

```shell
dyld[42223]: Library not loaded: /opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib
  Referenced from: <8648640B-CE33-3DF9-8EC7-34E841FCE43F> /opt/local/bin/sunshine-0.19.1
  Reason: tried: '/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (no such file), '/opt/local/libexec/boost/1.76/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libboost_locale-mt.dylib' (no such file), '/usr/lib/libboost_locale-mt.dylib' (no such file, not in dyld cache)
zsh: abort      sunshine
```

**Comment by ReenigneArcher** [comment]
You need to use macports and the `Portfile`

**Comment by MELKOHEN** [comment]
I just did it using the portfile, it worked for the installation, but here's the error message I'm getting: dyld[46619]: Library not loaded: /opt/local/lib/libopus.0.dylib
Referenced from: <8648640B-CE33-3DF9-8EC7-34E841FCE43F> /opt/local/bin/sunshine-0.19.1
Reason: tried: '/opt/local/lib/libopus.0.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/lib/libopus.0.dylib' (no such file), '/opt/local/lib/libopus.0.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libopus.0.dylib' (no such file), '/usr/lib/libopus.0.dylib' (no such file, not in dyld cache)
zsh: abort sunshine

**Comment by MELKOHEN** [comment]
I've installed libopus +universal, then I tried for boost but I get this : 
% sunshine
dyld[42921]: Library not loaded: /opt/local/libexec/boost/1.80/lib/libboost_locale-mt.dylib
  Referenced from: <8648640B-CE33-3DF9-8EC7-34E841FCE43F> /opt/local/bin/sunshine-0.19.1
  Reason: tried: '/opt/local/libexec/boost/1.80/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/libexec/boost/1.80/lib/libboost_locale-mt.dylib' (no such file), '/opt/local/libexec/boost/1.80/lib/libboost_locale-mt.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libboost_locale-mt.dylib' (no such file), '/usr/lib/libboost_locale-mt.dylib' (no such file, not in dyld cache)
zsh: abort      sunshine

**Comment by ReenigneArcher** [comment]
When you install using our Portfile, it should automatically get dependencies for your architecture. https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/installation.html#portfile

Also the dmg was not supposed to be published as we know it doesn't work. I've removed that release asset.

Can you open a post on our discord so we can discuss this there?

**Comment by MELKOHEN** [comment]
Thank You for your answer Reenigne Archer. 

I've used initially the dmg file, I'm trying to install everything and retry. 

ok for discord !

**Reply by samuelhe52** [reply]
Have you resolved your problem? I'm seeing the same terminal output on my mac and would be truly grateful if someone could help me out.

**Reply by samuelhe52** [reply]
I checked discord for your post and didn't find any. Could you post a link here?

**Reply by vukael39** [reply]
I have same error when install Sunshine via Portfile (both version 0.20 and 0.19.1). Library files error - incompatible architecture. Checked discord but cannot find solution. Can @MELKOHEN and @samuelhe864 help if you made it worked. Thank you.

[a comment by an automated account is not shown]
